v4/test/auth/auth_mock_test.go

```diff
@@ -0,0 +1,267 @@
+				w.Write([]byte(fmt.Sprint(`{"access_token": "` + AccessToken + `", "expires_in": "5"}`)))
+			})
+			sToken := httptest.NewServer(muxToken)
+			defer sToken.Close()
+
+			// provides all endpoints
+			muxEndpoints := http.NewServeMux()
+			muxEndpoints.HandleFunc("/endpoints", func(w http.ResponseWriter, r *http.Request) {
+				w.Write([]byte(fmt.Sprintf(`{"token_endpoint": "` + sToken.URL + `/auth"}`)))
+			})
+			sEndpoints := httptest.NewServer(muxEndpoints)
+			defer sEndpoints.Close()
+
+			// Returns the address of the auth server
+			mux := http.NewServeMux()
+			mux.HandleFunc("/v1/.well-known/openid-configuration", func(w http.ResponseWriter, r *http.Request) {
+				w.Write([]byte(`{"href": "` + sEndpoints.URL + `/endpoints", "clientId": "DoesNotMatter"}`))
+			})
+			mux.HandleFunc("/v1/schema", func(w http.ResponseWriter, r *http.Request) {
+				w.Write([]byte(`{}`))
+			})
+			s := httptest.NewServer(mux)
+			defer s.Close()
+
+			cfg, err := weaviate.NewConfig(strings.TrimPrefix(s.URL, "http://"), "http", tc.authConfig, nil)
+			assert.Nil(t, err)
+			assert.True(t, strings.Contains(buf.String(), "Auth002"))
+
+			client := weaviate.New(*cfg)
+			AuthErr := client.Schema().AllDeleter().Do(context.TODO())
+			assert.Nil(t, AuthErr)
+		})
+	}
+}
+
+// Test that client using CC automatically get a new token after expiration
+func TestAuthMock_RefreshCC(t *testing.T) {
+	i := 0
+	// endpoint for access tokens
+	muxToken := http.NewServeMux()
+	muxToken.HandleFunc("/auth", func(w http.ResponseWriter, r *http.Request) {
+		i += 1 // record how often this was called
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte(fmt.Sprint(`{"access_token": "` + AccessToken + `", "expires_in": "1"}`)))
+	})
+	sToken := httptest.NewServer(muxToken)
+	defer sToken.Close()
+
+	// provides all endpoints
+	muxEndpoints := http.NewServeMux()
+	muxEndpoints.HandleFunc("/endpoints", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(fmt.Sprintf(`{"token_endpoint": "` + sToken.URL + `/auth"}`)))
+	})
+	sEndpoints := httptest.NewServer(muxEndpoints)
+	defer sEndpoints.Close()
+
+	// Returns the address of the auth server
+	mux := http.NewServeMux()
+	mux.HandleFunc("/v1/.well-known/openid-configuration", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`{"href": "` + sEndpoints.URL + `/endpoints", "clientId": "DoesNotMatter"}`))
+	})
+	mux.HandleFunc("/v1/schema", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`{}`))
+	})
+	s := httptest.NewServer(mux)
+	defer s.Close()
+
+	cfg, err := weaviate.NewConfig(strings.TrimPrefix(s.URL, "http://"), "http", auth.ClientCredentials{ClientSecret: "SecretValue"}, nil)
+	assert.Nil(t, err)
+	client := weaviate.New(*cfg)
+	AuthErr := client.Schema().AllDeleter().Do(context.TODO())
+	assert.Nil(t, AuthErr)
+	assert.Equal(t, i, 3) // client does 3 initial calls to token endpoint
+
+	time.Sleep(time.Second * 5)
```

Comment:
I would suggest to change this gouroutine and refactor it using a `ticker` not `sleeps`